paasta_tools/autoscaling_lib.py

```diff
@@ -664,23 +649,38 @@ def wait_and_terminate(slave, drain_timeout, dry_run, region=None):
                 raise
 
 
+def sort_slaves_to_kill(slaves):
+    """Pick the best slaves to kill. This returns a list of slaves
+    after sorting in preference of which slaves we kill first.
+    It sorts first by number of chronos tasks, then by total number of tasks
+
+    :param slaves: list of slaves dict
+    :returns: list of slaves dicts"""
+    if slaves:
+        slaves_by_task_counts = [slave for slave in
+                                 sorted(slaves, key=lambda x: (x['task_counts'].chronos_count, x['task_counts'].count))]
+        return slaves_by_task_counts
+    else:
+        return []
+
+
 def get_instance_type_weights(sfr):
     launch_specifications = sfr['SpotFleetRequestConfig']['LaunchSpecifications']
     return {ls['InstanceType']: ls['WeightedCapacity'] for ls in launch_specifications}
 
 
-def filter_sfr_slaves(sorted_slaves, resource):
+def filter_sfr_slaves(slaves_list, resource):
```

Comment:
I think the plural is ok here. It's a list of slaves rather than a list of things about a slave.